src/balatro_gym/game/shop.py

```diff
@@ -0,0 +1,61 @@
+import dataclasses
+import random
+from typing import Sequence
+
+from balatro_gym.cards.interfaces import Card
+from balatro_gym.cards.voucher import ALL_VOUCHERS
+from balatro_gym.interfaces import BoosterPack, Voucher
+
+
+@dataclasses.dataclass
+class ShopState:
+    buyable_cards: Sequence[Card]
+    vouchers: Sequence[Voucher]
+    booster_packs: Sequence[BoosterPack]
+
+
+class Shop:
+
+    def __init__(self,
+    num_buyable_slots: int = 2,
+    num_vouchers: int = 1,
+    num_booster_packs: int = 2,
+    reroll_price: int = 5):
+        self.num_buyable_slots = num_buyable_slots
+        self.num_vouchers = num_vouchers
+        self.num_booster_packs = num_booster_packs
+        self.reroll_price = reroll_price
+        self.vouchers: Sequence[Voucher] = []
+        self.bought_vouchers: Sequence[Voucher] = []
+        self.current_state = None
+
+    def increase_num_buyable_slots(self) -> None:
+        self.num_vouchers += 1
+
+    def increase_voucher_limit(self) -> None:
+        self.num_vouchers += 1
+
+    def change_reroll_price(self, price: int) -> None:
+        self.reroll_price = price
+
+    def reroll(self) -> ShopState:
+        return ShopState([], self.vouchers, [])
+
+    def voucher_generator(self) -> Sequence[Voucher]:
+        valid_vouchers = []
+        for voucher in ALL_VOUCHERS:
```

Comment:
Nit: Rather than iterating over `ALL_VOUCHERS` and checking `if ... not in self.bought_vouchers` on L51, you could iterate over the set `ALL_VOUCHERS.difference(self.bought_vouchers)`.

Both your condition in L51 and my suggestion above assumes that vouchers can only be purchased once, which I think is true? The [wiki](https://balatrogame.fandom.com/wiki/Vouchers) states
```
No Voucher can be claimed twice in a single run normally
```
which sort of leaves room for interpretation.